doc/sai-ptf/tunnel_test_plan.md

```diff
@@ -0,0 +1,717 @@
+
+    We will send encapsulated packet from lag2 and expect a decapsulated packet on port1
+    -----------------------------------------------------------------
+    Egress side[port1]           |          ingress side[lag2]
+    ------------------------------------------------------------------
+    ipv4's falls in 192.168.1.0     |        ipv4's falls in 10.1.0.0
+    ------------------------------------------------------------------
+    ipv6's falls in fc02::   |   ipv6's falls in fc00:1::
+    ------------------------------------------------------------------
+
+### Testing Data Packet
+- PIPE MODE PACKET:
+#### IPV4 IN IPV4 Packet <!-- omit in toc --> 
+- ingress encap packet=Ether(dst=ROUTER_MAC)/IP(src=10.1.2.100,dst=10.10.10.1, ttl=64)/IP(src=192.168.20.1,dst=192.168.1.1, ttl=51)/TCP()
+- expected decap packet = Ether(dst=01:01:00:99:01:01,src=ROUTER_MAC)/IP(src=192.168.20.1,dst=192.168.1.1,ttl=50)/TCP()
+
+#### IPV6 IN IPV6 Packet <!-- omit in toc --> 
+- ingress encap packet=Ether(dst=ROUTER_MAC)/IP(src=fc00:1::2:100,dst=4001:0E98:03EE::0D25, ttl=64)/IP(src=2001:0000:25DE::CADE,dst=fc02::1:1, ttl=51)/TCP()
+- expected decap packet = Ether(dst=01:01:00:99:01:01,src=ROUTER_MAC)/IP(src=2001:0000:25DE::CADE,dst=fc02::1:1,ttl=50)/TCP()
+
+- UNIFORM MODE PACKET:
+#### IPV4 IN IPV4 Packet <!-- omit in toc --> 
+- ingress encap packet=Ether(dst=ROUTER_MAC)/IP(src=10.1.2.100,dst=10.10.10.1, ttl=64)/IP(src=192.168.20.1,dst=192.168.1.1, ttl=50)/TCP()
+- expected decap packet = Ether(dst=01:01:00:99:01:01,src=ROUTER_MAC)/IP(src=192.168.20.1,dst=192.168.1.1,ttl=63)/TCP()
+
+#### IPV6 IN IPV6 Packet <!-- omit in toc --> 
+- ingress encap packet=Ether(dst=ROUTER_MAC)/IP(src=fc00:1::2:100,dst=4001:0E98:03EE::0D25, ttl=64)/IP(src=2001:0000:25DE::CADE,dst=fc02::1:1, ttl=50)/TCP()
+- expected decap packet = Ether(dst=01:01:00:99:01:01,src=ROUTER_MAC)/IP(src=2001:0000:25DE::CADE,dst=fc02::1:1,ttl=63)/TCP()
+### Test steps: <!-- omit in toc --> 
+- decap_ttl_set_pipe_mod
+1. Set ipinip tunnel encap_ttl_mode attr with SAI_TUNNEL_TTL_MODE_PIPE_MODEL
+2. Generate input ipinip packet with ip_ttl field in outer ip header as 64 , one in inner ip header as 51, expected recieved packet with ip_ttl field as 50.
+3. Send packet from lag2.
+4. Recieve ipinip packet from port1, compare it with expected packet.
+
+- decap_ttl_set_uniform_mode
+1. Set ipinip tunnel encap_ttl_mode attr with with SAI_TUNNEL_TTL_MODE_UNIFORM_MODEL
+2. Generate input ipinip packet with ip_ttl field in outer ip header as 64 , inner ip_ttl as 50.
+3. Send packet from lag2 port.
+4. Create expected recieved packet with ip_ttl field as 63.
+5. Recieve ipinip packet from port1, compare it with expected packet.
+
+## Test Group12: IP IN IP ENCAP DSCP 
+	
+### Case1: encap_dscp_set_pipe_mode_v4
+### Case2: encap_dscp_set_pipe_mode_v6
+### Case3: encap_dscp_set_uniform_mode_v4
+### Case4: encap_dscp_set_uniform_mode_v6
+
+
```

Comment:
Do we need add some MTU related testing in tunnel, like total length over l3 MTU, check if right behavior happened. 
And different inner packet length during testing.